network-api/networkapi/wagtailpages/tests/buyersguide/test_homepage.py

```diff
@@ -297,3 +300,73 @@ def test_category_page_context_with_cta_enabled(self):
 
         self.assertEqual(response.status_code, 200)
         self.assertIsNotNone(response.context['featured_cta'])
+
+
+class BuyersGuideCallToActionTest(TestCase):
+
+    def setUp(self):
+        edit_handler = get_snippet_edit_handler(BuyersGuideCallToAction)
+        self.form_class = edit_handler.get_form_class()
+
+    def test_return_target_url_with_external_link(self):
+        test_url = "http://test.com"
+        cta = BuyersGuideCallToAction.objects.create(
+            title="Test CTA",
+            link_label="Test Link",
+            link_target_url=test_url,
+        )
+        cta.save()
+
+        target_url = cta.get_target_url()
+
+        self.assertEqual(target_url, test_url)
+
+    def test_return_target_url_with_page_link(self):
+        test_page = buyersguide_factories.BuyersGuidePageFactory()
+        cta = BuyersGuideCallToAction.objects.create(
+            title="Test CTA",
+            link_label="Test Link",
+            link_target_page=test_page,
+        )
+        cta.save()
+
+        target_url = cta.get_target_url()
+
+        self.assertEqual(target_url, test_page.url)
+
+    def test_cannot_be_missing_external_or_page_link(self):
+        form = self.form_class(
+            data=({"title": "Test CTA", "link_label": "Test Link"}),
+        )
+
+        self.assertFalse(form.is_valid())
+        self.assertEqual(2, len(form.errors))
+        self.assertIn("link_target_url", form.errors)
+        self.assertIn("link_target_page", form.errors)
+
+    def test_cannot_have_both_external_and_page_link(self):
+        test_page = buyersguide_factories.BuyersGuidePageFactory()
+        form = self.form_class(
+            data=(
+                {
+                    "title": "Test CTA",
+                    "link_label": "Test Link",
+                    "link_target_url": "http://test.com",
+                    "link_target_page": test_page,
+                }
+            )
+        )
+
+        self.assertFalse(form.is_valid())
+        self.assertEqual(2, len(form.errors))
+        self.assertIn("link_target_url", form.errors)
+        self.assertIn("link_target_page", form.errors)
+
+    def test_cannot_be_missing_link_label(self):
+        form = self.form_class(
+            data=({"title": "Test CTA", "link_target_url": "http://test.com"}),
+        )
+
+        self.assertFalse(form.is_valid())
+        self.assertEqual(1, len(form.errors))
+        self.assertIn("link_label", form.errors)
```

Comment:
Have you considered adding test for all the other possible combinations too? We have 3 independent boolean variables. That would mean 2 * 2 * 2 = 8 possible combinations we would need to test.

Reply:
Added!

Reply:
I think one test is duplicated and one is missed. This should be an easy fix 👍